pints/plot.py

```diff
@@ -212,6 +221,18 @@ def histogram(samples, kde=False, ref_parameters=None, n_percentiles=None):
     n_list = len(samples)
     _, n_param = samples[0].shape
 
+    # Check parameter names
+    if parameter_names is not None:
```

Comment:
I would do an `if parameter_names is None:` here that creates the default, and then `else` this code.
Then the second `if` inside the loop can go

Reply:
I'm note sure I understand what you mean by loop here. Do you mean something like
```
if parameter_name is None:
    pass
elif len(parameter_names) != n_params:
   raise ....
```

Reply:
But I can revert the order in the plotting loops to 
```
if parameter_name is None:
   ... enumerate params ...
else:
   ... get names from parameter_names  ...
```

Reply:
I meant getting rid of the `if` further down, when you're using the names. If you just create a `parameter_names` here you don't need to have an `if` in the plotting loop

Reply:
```
if parameter_names is None:
    parameter_names = <generate default names>
else:
    <check user input>

...

for i in ..
  for j in ...
     plot(..., parameter_names[0]
```

Reply:
I see, that makes sense!